sentry-android-core/src/main/java/io/sentry/android/core/internal/gestures/SentryGestureListener.java

```diff
@@ -18,25 +24,35 @@
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.VisibleForTesting;
 
 @ApiStatus.Internal
 public final class SentryGestureListener implements GestureDetector.OnGestureListener {
 
-  private final @NotNull WeakReference<Window> windowRef;
+  public static final String UI_ACTION = "ui.action";
+
+  private final @NotNull WeakReference<Activity> activityRef;
   private final @NotNull IHub hub;
   private final @NotNull SentryAndroidOptions options;
   private final boolean isAndroidXAvailable;
+  private final @NotNull ActivityFramesTracker activityFramesTracker;
+
+  private @NotNull WeakReference<View> activeView = new WeakReference<>(null);
```

Comment:
Shouldn't be initing once there's a view? initing with `null` is suboptimal maybe?

Reply:
mm, I just wanted to avoid null check for the weakref itself, but yeah would be better to init it when there's a view, true 👍